juju/juju
# CLI Command for "show-generation"

## Description of change

This patch adds a new CLI command, _show-generation_, which displays a summary of the "next" generation consisting of applications with changes, units advanced to the generation, and configuration changes applied to the generation.

## QA steps

- `export JUJU_DEV_FEATURE_FLAGS=generations`
- Bootstrap.
- `juju deploy redis -n 2`
- `juju add-generation`
- `juju advance-generation redis/0`
- `juju config redis databases=8`
- `juju show-generation`
- Output should match:
```yaml
next:
- application: redis
  units:
  - redis/0
  config:
    databases: 8
```

## Documentation changes

Yes, this CLI command should be documented with the generations feature generally.

## Bug reference

N/A
